### PR TITLE
Tweak expiry metadata

### PR DESCRIPTION
**Pull request by mattandrews**
Two things here:

1. Log in Datadog the application status (eg. `PENDING`, `COMPLETE` etc) when we delete expired apps. This will give us some (potentially painful) data on what state these apps are in at expiry time.

2. Log the `userId` in the email queue table. This is to aid debugging in the case where a customer gets in touch after their application has expired but we can't prove it was deleted as we don't know its application ID.

### CODE REVIEW

File: db/models/applications.js [hunk 1]
```diff
@@ -372,6 +372,17 @@ class ApplicationEmailQueue extends Model {
...
+            /**
+             * User model reference
+             * (optional as it was added after the model was created)
+             * Intended to allow looking up a user's deleted applications
+             * after the application itself was deleted
+             */
+            userId: {
+                type: DataTypes.INTEGER,
+                allowNull: true
+            },
```
Comment (mattandrews) [inline db/models/applications.js]: My plan is to just manually add this column to the test/live tables – given it allows NULL it's a safe change (just done it locally) even before we merge this.

**Comment by mattandrews** [comment]
[Store the userID alongside the applicationID in the application expiry emails table (to aid debugging)](https://trello.com/c/GjAHgX3Z/1016-store-the-userid-alongside-the-applicationid-in-the-application-expiry-emails-table-to-aid-debugging)
